examples/gno.land/p/moul/md/md.gno

```diff
@@ -299,3 +300,146 @@ func ColumnsN(content []string, colsPerRow int, padded bool) string {
+	FormEmail
+
+	// FormTel represents a telephone number input.
+	// Example: "+33 6 12 34 56 78"
+	FormTel
+
+	// FormPassword represents a password input.
+	// The value is hidden from display.
+	FormPassword
+)
+
+var input2str = map[FormInputType]string{
+	FormText:     "text",
+	FormNumber:   "number",
+	FormEmail:    "email",
+	FormTel:      "tel",
+	FormPassword: "password",
+}
+
+// FormInput represents a single form input field.
+// Name is the input name, PlaceHolder is the placeholder text,
+// and Type determines the input type (text, number, etc.).
+type FormInput struct {
+	Name        string
+	PlaceHolder string
+	Type        FormInputType
+}
+
+// Form represents an entire form block with a path and multiple input fields.
+// The form can be rendered as a string in a structured format.
+type Form struct {
+	Path   string
+	Inputs []FormInput
+}
+
+// SetPath defines the form submission path.
+// This path determines the endpoint to which the form will send data.
+// If unset, the form uses the current page path.
+func (f *Form) SetPath(path string) {
+	f.Path = path
+}
+
+func (f *Form) addInput(t FormInputType, n, p string) {
+	f.Inputs = append(f.Inputs, FormInput{
+		Name:        n,
+		PlaceHolder: p,
+		Type:        t,
+	})
+}
+
+// AddInput adds a generic text input to the form.
+// Equivalent to AddTextInput.
+func (f *Form) AddInput(name, placeholder string) {
+	f.addInput(FormText, name, placeholder)
+}
+
+// AddTextInput adds a text input field to the form.
+// Example: a string like "hello"
+func (f *Form) AddTextInput(name, placeholder string) {
+	f.addInput(FormText, name, placeholder)
+}
+
+// AddNumberInput adds a numeric input field to the form.
+// Example: an integer like 42
+func (f *Form) AddNumberInput(name, placeholder string) {
+	f.addInput(FormNumber, name, placeholder)
+}
+
+// AddEmailInput adds an email input field to the form.
+// The email must match the standard email format.
+// Example: "example@domain.com"
+func (f *Form) AddEmailInput(name, placeholder string) {
+	f.addInput(FormEmail, name, placeholder)
+}
+
+// AddTelInput adds a telephone number input field to the form.
+// Example: "+33 6 12 34 56 78"
+func (f *Form) AddTelInput(name, placeholder string) {
+	f.addInput(FormTel, name, placeholder)
+}
+
+// AddPasswordInput adds a password input field to the form.
+// The input is visually hidden during typing.
+func (f *Form) AddPasswordInput(name, placeholder string) {
+	f.addInput(FormPassword, name, placeholder)
+}
+
+// String returns the form as a Gno-formatted string.
+// It generates the <gno-form> block with all included inputs.
+// Example output:
+// <gno-form path="/submit">
+//   <gno-input name="email" placeholder="Your email" type="email" />
+// </gno-form>
+func (f Form) String() string {
+	var sb strings.Builder
+
+	// Write the begin of the form
+	str := "<gno-form"
+	if f.Path != "" {
+		str += " path=\"" + EscapeText(f.Path) + "\""
```

Comment:
You should not use EscapeText but another Escape, probably EscapeValue; EscapeText is designed for common text, like a paragraph.